### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by nsantos16**
### Description
As a user I should be able to sign up so my account is created in Target MVD.

### Preview
screen shot 2019-02-13 at 3 44 21 pm

### CODE REVIEW

File: src/actions/userActions.js [hunk 1]
```diff
@@ -40,8 +40,9 @@ export const logout = () =>
 export const signUp = user =>
   async () => {
     try {
+      user.password_confirmation = user.password;
       const response = await userApi.signUp({ user });
-      sessionService.saveUser(response.user);
+      sessionService.saveUser(response);
```
Comment (agustito37) [inline src/actions/userActions.js]: Alo!!!! Mmm I think you need save the user that's inside the response, why did you change this?.

File: src/components/App.js [hunk 2]
```diff
@@ -6,6 +6,8 @@ import { sessionService } from 'redux-react-native-session';
 import configureStore from 'store/configureStore';
 import registerScreens from '../screens';
 
+import * as consts from '../utils/constants';
```
Comment (agustito37) [inline src/components/App.js]: There's an utils shortcut. Just `'utils/constants'`. You can see them in the `.babelsrc` file.

File: src/components/common/Button/index.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { View, TouchableOpacity, Text } from 'react-native';
+import * as consts from 'utils/constants';
```
Comment (agustito37) [inline src/components/common/Button/index.js]: Maybe a better option is to destructure this object as we are doing in the line above.

File: src/components/common/Button/index.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import styles from './styles';
+
+const Button = ({ title, color = 'black', onPress, size = 'lg' }) => (
+  <View style={styles(size, color).container}>
```
Comment (agustito37) [inline src/components/common/Button/index.js]: You are calling the styles function every time this component renders so, a better option here is to define save the styles(size, color) inside a variable and then destructure the values you want.  
```
const Button = ({ title, color = 'black', onPress, size = 'lg' }) => {
const { container, btnFont } = styles(size, color);
return (...);
};
```

File: src/components/common/Button/index.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      style={styles(size, color).container}
+    >
+      <Text adjustsFontSizeToFit style={[consts.btnFont, styles(size, color).btn]}>
+        {title}
+      </Text>
+    </TouchableOpacity>
+  </View>
+);
+
+const { oneOf, string, func } = PropTypes;
```
Comment (agustito37) [inline src/components/common/Button/index.js]: You can destructure this at the top of the file when you import the PropTypes.

File: src/components/common/Button/styles.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { StyleSheet } from 'react-native';
+import * as consts from 'utils/constants';
```
Comment (agustito37) [inline src/components/common/Button/styles.js]: Same here, destructure is life.

File: src/components/common/Input/index.js [hunk 7]
```diff
@@ -1,30 +1,63 @@
...
 
-const Input = ({ input: { onChange, ...restInput }, password = false, label, meta: { touched, error } }) => (
+class TextInput2 extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { placeholder: props.value.length == 0 };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(ev) {
```
Comment (agustito37) [inline src/components/common/Input/index.js]: A minor comment you can avoid constructor binding every time you define a function if you use e.g:
`handleChange = (ev) => {}` the arrow function is going to bind the method to the context.

File: src/components/common/Input/index.js [hunk 8]
```diff
@@ -1,30 +1,63 @@
...
+    const { placeholder } = this.state;
+    return <TextInput
+      {...rest}
+      onChange={this.handleChange}
+      style={placeholder ? [style, placeholderStyle] : style}
+    />;
+  }
+}
+
+const Input = ({ input: { onChange, ...restInput }, size = 'lg', password = false, label, maxLength = 80, placeHolderText = '', meta: { touched, error } }) => (
```
Comment (agustito37) [inline src/components/common/Input/index.js]: Can you break this line? is too long.

File: src/actions/userActions.js (hunk 1 above)
Comment (nsantos16) [reply src/actions/userActions.js]: I changed because the response of API is in the response itself, not in response.user .

Comment (agustito37) [reply src/actions/userActions.js]: Cool!

File: src/components/common/Input/index.js [hunk 9]
```diff
@@ -1,30 +1,66 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { View, TextInput, Text } from 'react-native';
+import { placeholderFont, brownGrey } from 'utils/constants';
 import styles from './styles';
 
-const Input = ({ input: { onChange, ...restInput }, password = false, label, meta: { touched, error } }) => (
-  <View>
-    {label && <Text>{label}</Text>}
+class TextInput2 extends React.Component {
```
Comment (missx) [inline src/components/common/Input/index.js]: Can you think of a more semantic name than TextInput2 for the component?

File: src/components/common/Input/index.js [hunk 10]
```diff
@@ -1,30 +1,66 @@
...
+      </View>
       {touched && error && <Text>{error}</Text>}
     </View>
-  </View>
-);
+  );
+};
 
-const { string, object, bool } = PropTypes;
+const { oneOf, string, object, bool, number } = PropTypes;
```
Comment (missx) [inline src/components/common/Input/index.js]: You could just import these types from proptypes instead of importing the whole library, like so: ```import { oneOf, string, object, bool, number } from 'prop-types';```

File: src/components/common/Input/index.js (hunk 9 above)
Comment (missx) [inline src/components/common/Input/index.js]: You should also add proptypes to this component.

File: src/components/common/Line/index.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { oneOf, number } from 'prop-types';
+import { View } from 'react-native';
+import styles from './styles';
+
+const Input = ({ size, padding }) => {
```
Comment (missx) [inline src/components/common/Line/index.js]: Shouldn't this component be named 'Line'? Or is it in the wrong folder?

File: src/components/user/LoginForm/index.js [hunk 12]
```diff
@@ -13,7 +13,7 @@ const LoginForm = ({ handleSubmit, error }) => (
     {error && <Text>{error}</Text>}
     <Field
       name="email"
-      label={translate('SIGN_IN.email')}
+      label="changed"
```
Comment (missx) [inline src/components/user/LoginForm/index.js]: Why is the label 'changed'?

File: src/components/common/Input/index.js (hunk 9 above)
Comment (felipegarcia92) [reply src/components/common/Input/index.js]: Also, try to keep the component and the file/directory name equal. And why not separate this component into a new file? One component per file is OK I think.

File: src/components/user/SignUpForm/index.js [hunk 13]
```diff
@@ -1,44 +1,59 @@
...
   </View>
 );
 
-const { func, string } = PropTypes;
-
 SignUpForm.propTypes = {
   handleSubmit: func.isRequired,
-  error: string
+  error: string,
+  navigator: object
```
Comment (missx) [inline src/components/user/SignUpForm/index.js]: Maybe navigator should be required, don't you think?

File: src/containers/SignUpScreen/index.js [hunk 14]
```diff
@@ -1,29 +1,26 @@
...
+    <SignUpForm onSubmit={user => signUp(user.toJS())} navigator={navigator} />
+  </ImageBackground>
 );
 
-const { func } = PropTypes;
-
 SignUpScreen.propTypes = {
-  signUp: func.isRequired
+  signUp: func.isRequired,
+  navigator: object
```
Comment (missx) [inline src/containers/SignUpScreen/index.js]: Here too.

File: src/components/common/Input/index.js [hunk 15]
```diff
@@ -1,30 +1,66 @@
...
+    const { placeholder } = this.state;
+    return <TextInput
+      {...rest}
+      onChange={this.handleChange}
+      style={placeholder ? [style, placeholderStyle] : style}
+    />;
+  }
+}
+
+const Input = ({ input: { onChange, ...restInput }, size = 'lg', password = false,
```
Comment (felipegarcia92) [inline src/components/common/Input/index.js]: Normally we prefer separating them with new lines:
```
const Input = ({ 
  input: { onChange, ...restInput },
  size = 'lg',
  password = false,
... etc
```

File: src/components/common/Line/index.js (hunk 11 above)
Comment (felipegarcia92) [inline src/components/common/Line/index.js]: The component name doesn't match the file name. Can you rename it?

File: src/components/user/SignUpForm/index.js [hunk 16]
```diff
@@ -1,44 +1,59 @@
...
     />
-    <View style={styles.button}>
-      <Button title={translate('SIGN_UP.button')} onPress={handleSubmit} />
+    <View style={styles.signUp}>
+      <Button title={translate('SIGN_UP.signUp')} onPress={handleSubmit} />
+    </View>
+    <Line size="lg" padding={19} />
+    <View style={styles.signIn}>
+      <Button
+        color="white" title={translate('SIGN_IN.title')} onPress={() => navigator.pop()}
```
Comment (felipegarcia92) [inline src/components/user/SignUpForm/index.js]: `() => navigator.pop()` equals `navigation.pop`, so i'd use the shortest :)

File: src/components/user/LoginForm/index.js (hunk 12 above)
Comment (nsantos16) [reply src/components/user/LoginForm/index.js]: I have changed it only for tested, I forgot to change it.

File: src/components/user/SignUpForm/index.js (hunk 16 above)
Comment (nsantos16) [reply src/components/user/SignUpForm/index.js]: If you invoking navigator.pop, the function is not binding by it self.
